pkg/apis/application/v1alpha1/types.go

```diff
@@ -1287,6 +1287,9 @@ func (in RevisionHistories) LastRevisionHistory() RevisionHistory {
 
 // Trunc truncates the list of history items to size n
 func (in RevisionHistories) Trunc(n int) RevisionHistories {
+	if n < 0 {
```

Comment:
although a smallish change, we should add some test cases

Reply:
I would expand on [Test_appStateManager_persistRevisionHistory](https://github.com/argoproj/argo-cd/blob/3baca9b6960691435e039dd756ae122a272ebd42/controller/state_test.go#L904) and add a case where "-1" becomes 0. It should be pretty easy to get 100% code coverage on your changes :)